resources/top_level_overview.markdown

```diff
@@ -0,0 +1,283 @@
+# MapReduce
+
+This page serves as an overview of what constitutes a MapReduce job.  We then
+look into what's going on under the hood relating to the Hadoop implementation
+of MapReduce.  The lower-level, implementation details realting to
+[How Hadoop Handles MapReduce](#how-hadoop-handles-mapreduce) will be covered
+in the __Roadmap__ section.
+
+### What is MapReduce?
+As per _Map Reduce: Simplified Data Processing on Large Clusters_:
+
+MapReduce is a programming model inspired by the synonymous primitives present
+in Lisp and many other functional programming languages.
+
+Conceptually, MapReduce can be broken into the following:
+
+1. __Computation__: Accepts a set of _input_ `(key, value)` pairs and produces a
+set of _output_ `(key, value)` pairs.
+
+2. __Map__: A user defined function that takes input pairs and produces a set of _intermediate_ `(key, value)` pairs. The MapReduce library groups all of these
+intermediate values associated with the same intermediate key, _I_, and passes
+them to the reduce function.
+
+3. __Reduce__:  Is a user defined function that accepts an intermediate key,
+`I`, and a set of values for that key. It merges the values together.  Typically
+zero or one output values are produced by any _Reduced_ invocation.
+(The list of values are passed by an iterator to deal with memory availability.)
```

Comment:
This sentence is unclear at this point, and perhaps too implementation-related anyway

Reply:
Cool, I'll go ahead and remove it.